src/OrderBookHandlers.ts

```diff
@@ -64,16 +64,18 @@ OrderBookContract.OrderChangeEvent.handler(({ event, context }) => {
 
   const maybeExistingOrder = context.SpotOrder.get(newSpotOrderChangeEvent.order_id);
   if (maybeExistingOrder) {
-    context.SpotOrder.set({
-      ...maybeExistingOrder,
-      base_size: newSpotOrderChangeEvent.new_base_size,
-      order_type:
-          eventOrder == null || eventOrder.base_size.value === 0n
-              ? undefined
-              : eventOrder.base_size.negative
-                  ? "sell"
-                  : "buy",
-    });
+    if (maybeExistingOrder.order_type != undefined) { // do not re-open an alreday closed order  
```

Comment:
It won't re-open an order.

Do you ever want to allow re-opening an order once it is closed? 
- I assume not - that would just be a new order with different new id. Right?

Reply:
> It won't re-open an order.
> 
> 
> 
> Do you ever want to allow re-opening an order once it is closed? 
> 
> - I assume not - that would just be a new order with different new id. Right?

order id is generated based on trader, base token and base price, i.e. a closed order can be reopened with the same order id

Reply:
Ok, will check if the fuel graphql interface is returning the data in the incorrect order.

Reply:
@JasoonS 
https://github.com/compolabs/orderbook-contract/blob/a209bd8ea59a1250eade1732b7c6b4ed90806fe5/contract/src/main.sw#L426